### ISSUE TITLE
bash run/test_agent.bash——Regarding the issue of test set results

### ISSUE THREAD

**Issue by xixizai29** [YanyuanQiao/HOP-VLN-finetune#3]
Hello, your work is excellent. I have encountered some minor issues. The results of the training set are close to those of your paper, but after training, I ran the test set code bash run/test_agent.bash, but the results were very poor. Do you know what the reason is? I look forward to your reply. The specific output is as follows:
root@04294b505380 :~/mount/Matterport3DSimulator# bash run/test_agent.bash
Optimizer: Using AdamW
Namespace(IMAGENET_FEATURES='img_features/ResNet-152-imagenet.tsv',  angle_feat_size=128, aug=None, aug_seen=None, aug_unseen=None, batchSize=8, description='VLNBERT-train-Prevalent', dropout=0.5, epsilon=0.1, featdropout=0.4, feature_size=2048, features='places365', feedback='sample', gamma=0.9, ignoreid=-100, iters=300000, load='snap/best_val_unseen', loadOptim=False, log_dir='snap/VLNBERT-train-Prevalent', lr=1e-05, maxAction=15, maxInput=80, ml_weight=0.2, name='VLNBERT-train-Prevalent', normalize_loss='total', optim='adamW', optimizer=<class 'torch.optim.adamw.AdamW'>, submit='True', teacher='final', teacher_weight=1.0, test_only=0, train='validlistener', vlnbert='prevalent', weight_decay=0.0, zero_init=False)

Start loading the image feature ... (~50 seconds)
Finish Loading the image feature from img_features/ResNet-152-places365.tsv in 22.5478 seconds
The feature size is 2048
Loading navigation graphs for 61 scans
R2RBatch loaded with 14039 instructions, using splits: train
The feature size is 2048
Loading navigation graphs for 18 scans
R2RBatch loaded with 4173 instructions, using splits: test

Initalizing the VLN-BERT model ...
you have loaded model from load_model/checkpoint
Loaded the listener model at iter 121000 from snap/best_val_unseen
result length 4173
Env name: test, nav_error: 9.2710, oracle_error: 0.0000, steps: 6.5950, lengths: 13.3260, success_rate: 0.0400, oracle_rate: 1.0000, spl: 0.0000

**Comment by 54cay** [comment]
I'm having the same issue:
root@d4d721669161:~/mount/Matterport3DSimulator/HOP-VLN-finetune/finetune_r2r# bash run/test_agent.bash
Optimizer: Using AdamW
Namespace(IMAGENET_FEATURES='img_features/ResNet-152-imagenet.tsv', angle_feat_size=128, aug=None, aug_seen=None, aug_unseen=None, batchSize=8, description='VLNBERT-train-Prevalent', dropout=0.5, epsilon=0.1, featdropout=0.4, feature_size=2048, features='places365', feedback='sample', gamma=0.9, ignoreid=-100, iters=300000, load='snap/VLNBERT-train-Prevalent/state_dict/best_val_unseen', loadOptim=False, log_dir='snap/VLNBERT-train-Prevalent', lr=1e-05, maxAction=15, maxInput=80, ml_weight=0.2, name='VLNBERT-train-Prevalent', normalize_loss='total', optim='adamW', optimizer=<class 'torch.optim.adamw.AdamW'>, submit='True', teacher='final', teacher_weight=1.0, test_only=0, train='validlistener', vlnbert='prevalent', weight_decay=0.0, zero_init=False)

Start loading the image feature ... (~50 seconds)
Finish Loading the image feature from img_features/ResNet-152-places365.tsv in 38.4631 seconds
The feature size is 2048
Loading navigation graphs for 61 scans
R2RBatch loaded with 14039 instructions, using splits: train
The feature size is 2048
Loading navigation graphs for 18 scans
R2RBatch loaded with 4173 instructions, using splits: test

Initalizing the VLN-BERT model ...
you have loaded model from load_model/checkpoint
Loaded the listener model at iter 220000 from snap/VLNBERT-train-Prevalent/state_dict/best_val_unseen
result length 4173
Env name: test, nav_error: 8.6982, oracle_error: 0.0000, steps: 6.0879, lengths: 12.2553, success_rate: 0.0388, oracle_rate: 1.0000, spl: 0.0000

**Comment by YanyuanQiao** [comment]
> Hello, your work is excellent. I have encountered some minor issues. The results of the training set are close to those of your paper, but after training, I ran the test set code bash run/test_agent.bash, but the results were very poor. Do you know what the reason is? I look forward to your reply. The specific output is as follows: root@04294b505380 :~/mount/Matterport3DSimulator# bash run/test_agent.bash Optimizer: Using AdamW Namespace(IMAGENET_FEATURES='img_features/ResNet-152-imagenet.tsv', angle_feat_size=128, aug=None, aug_seen=None, aug_unseen=None, batchSize=8, description='VLNBERT-train-Prevalent', dropout=0.5, epsilon=0.1, featdropout=0.4, feature_size=2048, features='places365', feedback='sample', gamma=0.9, ignoreid=-100, iters=300000, load='snap/best_val_unseen', loadOptim=False, log_dir='snap/VLNBERT-train-Prevalent', lr=1e-05, maxAction=15, maxInput=80, ml_weight=0.2, name='VLNBERT-train-Prevalent', normalize_loss='total', optim='adamW', optimizer=<class 'torch.optim.adamw.AdamW'>, submit='True', teacher='final', teacher_weight=1.0, test_only=0, train='validlistener', vlnbert='prevalent', weight_decay=0.0, zero_init=False)
> 
> Start loading the image feature ... (~50 seconds) Finish Loading the image feature from img_features/ResNet-152-places365.tsv in 22.5478 seconds The feature size is 2048 Loading navigation graphs for 61 scans R2RBatch loaded with 14039 instructions, using splits: train The feature size is 2048 Loading navigation graphs for 18 scans R2RBatch loaded with 4173 instructions, using splits: test
> 
> Initalizing the VLN-BERT model ... you have loaded model from load_model/checkpoint Loaded the listener model at iter 121000 from snap/best_val_unseen result length 4173 Env name: test, nav_error: 9.2710, oracle_error: 0.0000, steps: 6.5950, lengths: 13.3260, success_rate: 0.0400, oracle_rate: 1.0000, spl: 0.0000

Hi, sorry for the late reply!

This project was from about three years ago, so I don’t recall all the details clearly. But from what you shared, one possible issue is that the pretrained model might not have been successfully loaded.

You can try printing out the number of model parameters that were actually loaded when calling load_state_dict(), this can help confirm whether the checkpoint was properly restored.

Let me know if you need help with that.

**Comment by YanyuanQiao** [comment]
> I'm having the same issue: root@d4d721669161:~/mount/Matterport3DSimulator/HOP-VLN-finetune/finetune_r2r# bash run/test_agent.bash Optimizer: Using AdamW Namespace(IMAGENET_FEATURES='img_features/ResNet-152-imagenet.tsv', angle_feat_size=128, aug=None, aug_seen=None, aug_unseen=None, batchSize=8, description='VLNBERT-train-Prevalent', dropout=0.5, epsilon=0.1, featdropout=0.4, feature_size=2048, features='places365', feedback='sample', gamma=0.9, ignoreid=-100, iters=300000, load='snap/VLNBERT-train-Prevalent/state_dict/best_val_unseen', loadOptim=False, log_dir='snap/VLNBERT-train-Prevalent', lr=1e-05, maxAction=15, maxInput=80, ml_weight=0.2, name='VLNBERT-train-Prevalent', normalize_loss='total', optim='adamW', optimizer=<class 'torch.optim.adamw.AdamW'>, submit='True', teacher='final', teacher_weight=1.0, test_only=0, train='validlistener', vlnbert='prevalent', weight_decay=0.0, zero_init=False)
> 
> Start loading the image feature ... (~50 seconds) Finish Loading the image feature from img_features/ResNet-152-places365.tsv in 38.4631 seconds The feature size is 2048 Loading navigation graphs for 61 scans R2RBatch loaded with 14039 instructions, using splits: train The feature size is 2048 Loading navigation graphs for 18 scans R2RBatch loaded with 4173 instructions, using splits: test
> 
> Initalizing the VLN-BERT model ... you have loaded model from load_model/checkpoint Loaded the listener model at iter 220000 from snap/VLNBERT-train-Prevalent/state_dict/best_val_unseen result length 4173 Env name: test, nav_error: 8.6982, oracle_error: 0.0000, steps: 6.0879, lengths: 12.2553, success_rate: 0.0388, oracle_rate: 1.0000, spl: 0.0000

Hi! Thanks for your question.

This project was built a few years ago, so I don’t recall all the fine details. But based on the output you shared, one likely issue is that the pretrained model may not have been properly loaded.

You can try printing the number of parameters loaded using load_state_dict() — this can help verify whether the checkpoint was successfully restored. If the model wasn’t correctly initialized, it can lead to very poor performance like what you’re seeing.

Let me know if you need help checking this in the code.
